rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,89 @@
+
+## Why
+
+[Agentless EC2 discovery mode](https://github.com/gravitational/teleport/issues/17865) will discover and configure OpenSSH nodes so they can authenticate with a cluster. But those OpenSSH nodes aren't registered as `node` resources in the backend. We need a way to register agentless OpenSSH nodes as `node` resources so they can be viewed and managed by users. RBAC and session recording should function correctly with registered OpenSSH nodes as well.
+
+## Details
+
+### OpenSSH CA
+
+For security related reasons that will be discussed below, a new CA will be added called OpenSSH CA. The OpenSSH CA will be responsible for generating and signing user certificates that will be used to authenticate with registered OpenSSH nodes. The OpenSSH CA will not be used to generate certificates for existing or future *unregistered* OpenSSH nodes, preserving backwards compatibility for existing OpenSSH nodes.
+
+### RBAC
+
+When OpenSSH nodes are registered currently, RBAC checks for those nodes are not preformed. Even setting `auth_service.session_recording` to `proxy` in an Auth Server's config file does not help. RBAC logic will have to be updated so RBAC checks for registered OpenSSH nodes are preformed.
+
+When `(lib/srv.AuthHandlers).UserKeyAuth` is called to authenticate a node's user certificate, it will check what CA signed the certificate. If the OpenSSH CA signed the certificate, the node will be recognized as a registered OpenSSH node. `UserKeyAuth` will lookup the node's details and preform an RBAC check.
+
+### Registering nodes
+
+A new sub-kind to the `node` resource will be added for registered OpenSSH nodes: `openssh`. The absence of a `node` resource sub-kind will imply that a node is a Teleport agent node, making this change backwards compatible.
+
+#### Registering with `tctl`
+
+OpenSSH nodes can already be manually registered using `tctl create --force /path/to/node.yml`, though some changes should be made to make the process more straightforward for users. Agentless EC2 discovery mode will registered discovered OpenSSH nodes without needing user intervention, but if automatically registering an OpenSSH node fails a user may want to register a node manually. Currently nodes cannot be created with `tctl`, only upserted. This limitation should be removed so users can create nodes without having to pass `--force` to `tctl create`.
+
+Furthermore, `tctl` should not require as many fields to be set when creating nodes. This is an example `node` resource that will work with `tctl create` today:
+
+```yaml
+kind: node
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+`tctl create` will auto-generate `metadata.name` if it is not already set so users don't have to generate GUIDs themselves. Also, if `sub_kind` is set to `openssh`, `spec.public_addr` will not be allowed for registered OpenSSH nodes as it is not needed.
```

Comment:
nit:

> `tctl create` will auto-generate `metadata.name` if it is not already set so users don't have to generate GUIDs themselves.

I think this is not anything new. I'd just remove it TBH

Reply:
Currently if `metadata.name` is ommitted `tctl` will error with `missing parameter Name`